### PR TITLE
SSL integration for vault exporter

### PR DESCRIPTION
**Pull request by Cryson**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -12,6 +12,27 @@ import (
...
+		Default(":9410").String()
+	metricsPath = kingpin.Flag("web.telemetry-path",
+		"Path under which to expose metrics.").
+		Default("/metrics").String()
+	vaultCaCert = kingpin.Flag("vault-tls-cacert",
+		"The path to a PEM-encoded CA cert file to use to verify the Vault server SSL certificate.").
+		Envar("CONSUL_TLS_CACERT").String()
+	vaultClientCert = kingpin.Flag("vault-tls-client-cert",
+		"The path to the certificate for Vault communication.").
+		Envar("Vault_TLS_CLIENT_CERT").String()
```
Comment (latchmihay) [inline main.go]: please uppercase the Envvar *Vault*_TLS_CLIENT_CERT to *VAULT*_TLS_CLIENT_CERT

File: main.go [hunk 2]
```diff
@@ -12,6 +12,24 @@ import (
...
+		Default("/metrics").String()
+	vaultCaCert = kingpin.Flag("vault-tls-cacert",
+		"The path to a PEM-encoded CA cert file to use to verify the Vault server SSL certificate.").String()
+	vaultClientCert = kingpin.Flag("vault-tls-client-cert",
+		"The path to the certificate for Vault communication.").String()
+	vaultClientKey = kingpin.Flag("vault-tls-client-key",
+		"The path to the private key for Vault communication.").String()
+	sslInsecure = kingpin.Flag("insecure-ssl",
+		"Set SSL to ignore certificate validation.").
+			Default("false").Bool()
```
Comment (tomwilkie) [inline main.go]: Can you remove a level of indentation on this, to match the other, please?

File: main.go [hunk 3]
```diff
@@ -12,6 +12,24 @@ import (
...
 )
 
+var (
+	listenAddress = kingpin.Flag("web.listen-address",
+		"Address to listen on for web interface and telemetry.").
+		Default(":9410").String()
+	metricsPath = kingpin.Flag("web.telemetry-path",
+		"Path under which to expose metrics.").
+		Default("/metrics").String()
+	vaultCaCert = kingpin.Flag("vault-tls-cacert",
```
Comment (tomwilkie) [inline main.go]: This should be vaultCACert (as per https://github.com/golang/go/wiki/CodeReviewComments#initialisms).

File: main.go [hunk 4]
```diff
@@ -87,7 +126,7 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 		ch <- prometheus.MustNewConstMetric(
 			up, prometheus.GaugeValue, 0,
 		)
-		log.Errorf("Failted to collect health from Vault server: %v", err)
+		log.Errorf("Failed to collect health from Vault server: %v", err)
```
Comment (tomwilkie) [inline main.go]: :-) whoops thanks!

**Comment by tomwilkie** [comment]
Thanks for the contribution!

We've just been using the environment variables (ie VAULT_CACERT) to specify this kinda config (see https://www.vaultproject.io/docs/commands/index.html#environment-variables), but I don't mind putting this in when these couple of small nits are fixed.

**Pull request merged**